File: src/action/ActionShowCell.java
package com.leekwars.game.fight.action;

import com.alibaba.fastjson.JSONArray;
import com.leekwars.game.Util;
import com.leekwars.game.fight.entity.Entity;

public class ActionShowCell implements Action {

	private final int mLeek;
	private final int mCell;
	private final int mColor;

	public ActionShowCell(Entity l, int cell, int color) {
		mLeek = l.getFId();
		mCell = cell;
		mColor = color;
	}

	@Override
	public JSONArray getJSON() {
		JSONArray retour = new JSONArray();
		retour.add(Action.SHOW_CELL);
		retour.add(mLeek);
		retour.add(mCell);
		retour.add(Util.getHexaColor(mColor));

		return retour;
	}
}

File: src/action/ActionShowCell.cpp
#include "ActionShowCell.hpp"
#include "../util/Util.hpp"
#include "../entity/Entity.hpp"

ActionShowCell::ActionShowCell(Entity* e, int cell, int color) {
	this->leek = e->id;
	this->cell = cell;
	this->color = color;
}

Json ActionShowCell::json() const {
	return {(int) ActionType::SHOW_CELL, leek, cell, Util::int_to_hex_color(color)};
}
